src/lib/operators/join_hash.hpp

```diff
@@ -52,6 +52,13 @@ class JoinHash : public AbstractJoinOperator {
     size_t radix_bits{0};
     // Initially, the left input is the build side and the right side is the probe side.
     bool left_input_is_build_side{true};
+
```

Comment:
>I thought about adding these values to the visualization as well. But for now I have not, because:
> - a sole number is cumbersome, to calculate percentage (input table vs. materialized) we would need the input size, which we don't have

Why not?

> - this number works for the probe side and a "normal" join, where we can count the positions stored for the hash table
> - BUT I have no idea what to print for semi/anti joins; ideally we could compare the number of distinct values of the input table and the size of the hash table after the build phase, but we don't have the distinct count at hand

Have you considered simply tracking the number of values that didn't make it through the bloom filter? I'd hope for the cost of that to be minimal if we only synchronize at the end of the task.

Reply:
> Why not?

If I am not mistaken, the operator performance does not know it's parent when being printed. And we don't store the input tuple count. So I (i) have not found  a nice way to do it and (ii) don't think storing the input counts would be worth it.

> Have you considered simply tracking the number of values that didn't make it through the bloom filter? I'd hope for the cost of that to be minimal if we only synchronize at the end of the task.

Do you mean tracking it in the actual loops?

Reply:
> If I am not mistaken, the operator performance does not know it's parent when being printed. And we don't store the input tuple count. So I (i) have not found a nice way to do it and (ii) don't think storing the input counts would be worth it.

If, instead of storing absolute numbers, we calculate the bloom filter efficiency while writing the OperatorPerformanceData, we would still have the input sizes at hand.

> Do you mean tracking it in the actual loops?

Yes. Maybe that's worth testing.

Reply:
> If, instead of storing absolute numbers, we calculate the bloom filter efficiency while writing the OperatorPerformanceData, we would still have the input sizes at hand.

That's actually how I started. At had two floats like `build_side_bloom_reduction_ratio` which was in [0,1].
But I there is no way of having such a value for the semi case (we would need the distinct counts of the input table) and thus I found the current way is better (even though we might not be able to put it into the visualization).

> Yes. Maybe that's worth testing.

Really hesitant to introduce such a tracking in the most important operator. Might turn out to be fine on nemea, but we don't know about what is costs on AMD, ARM; we might only test the current page layout; and so on.
That's always true obviously, but for adding something in a hot loop, there needs to be a very good reason IMO. And if these counters are currently used by me three Hyrise users with very specific problems, I tend to not doing it.

Reply:
Fair.